docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+As the most prominent difference is that there are dedicated machines for the control plane components, we propose to notice the difference based on the existence of a worker pool for the control plane:
+
+```yaml
+spec:
+  provider:
+    workers:
+    - name: control-plane
+      controlPlane: {}
+```
+
+An alternative is to augment `.spec.controlPlane` with this information, however, we think adding a dedicated worker pool is the better and more natural fit (we need it anyways to enable re-use of most of the existing code).
+
+### Scenarios
+
+We distinguish between two different scenarios (for now):
+
+- [High Touch](#high-touch), meaning that there is no programmable infrastructure available.
+  We consider this the "bare metal" or "edge" use-case, where at first machines must be (often manually) prepared by human operators.
+  In this case, network setup or machine management is out of scope.
+- [Medium Touch](#medium-touch), meaning that there is programmable infrastructure available where we can leverage [provider extensions](../../extensions/README.md#infrastructure-provider) and [`machine-controller-manager`](https://github.com/gardener/machine-controller-manager) in order to manage the network setup and the machines.
+
+The general procedure of bootstrapping an autonomous shoot cluster is similar in both scenarios.
+
+#### High Touch
+
+In this scenario, the starting point is right on the first machine that shall be transformed into a control plane node.
+It is assumed that the network was set up properly (see [Prerequisites](#prerequisites)).
+Since machine management is out of scope, upgrades requiring to roll out the nodes (e.g., Kubernetes version upgrades, operating system upgrades, CA rotation, [see more](../usage/shoot_updates.md#rolling-update-triggers)) must be performed in-place.
+We will leverage the functionality that is to be developed in the scope of [#10219](https://github.com/gardener/gardener/issues/10219).
+This will also include operating system upgrades [planned for Garden Linux](https://github.com/gardener/gardener/issues/10219#issuecomment-2262299606).
+
+#### Medium Touch
+
+In this scenario, we can leverage the existing [`extensions.gardener.cloud/v1alpha1.Infrastructure` API](../extensions/infrastructure.md) in order to let the respective provider extension generate the necessary network resources (VPCs, subnets, etc., whatever is needed for the respective infrastructure).
+`machine-controller-manager` can be used to create the control plane nodes (and later also the worker nodes) of the autonomous shoot cluster.
+
+The challenge is that both Gardener provider extensions and `machine-controller-manager` depend on a pre-existing Kubernetes cluster or in general an API server.
+Hence, we propose to use a [KinD cluster](https://kind.sigs.k8s.io), deploy the needed components into it, and create the necessary `Infrastructure`, `MachineClass`, and `MachineDeployment` resources.
```

Comment:
Where is expected the kind cluster to run?

Reply:
There is no special assumption about the location of the KinD cluster. It should run somewhere where it can access the API of the target infrastructure. In a public cloud environment that means anywhere with internet access. In a private cloud environment, e.g. with openstack, the KinD cluster should be in a network being able to access the corresponding API, e.g. by being in the same infrastructure or having a VPN tunnel. It can run on the laptop of the operator if the network connectivity is ensured or it could be a virtual machine next to the target.